docs/sphinx/user_guide/feature/resource.rst

```diff
@@ -0,0 +1,247 @@
+single resource to the forall Indexset call. This resource will be used to execute the inner 
+execution loop of the indexset policy.::
+
+    using ExecPol = RAJA::ExecPolicy<RAJA::seq_segit, RAJA::cuda_exec<256>>;
+    RAJA::forall<ExecPol>(my_cuda_res, iset,  .... );
+
+When a resource is not provided by the user, a *default* resource is assigned from within the RAJA
+forall implementation. This default resource can be accessed in a number of ways.
+
+Directly from the concrete resource type::
+
+    RAJA::resources::Cuda my_default_cuda = RAJA::resources::Cuda::get_defualt();
+
+The resource type can be deduced from an execution policy::
+
+    using Res = RAJA::resources::get_resource<ExecPol>::type;
+    Res r = Res::get_defualt();
+
+Deduced from an execution policy and return the default directly::
+
+    auto my_resource = RAJA::resources::get_default_resource<ExecPol>();
+
+.. note:: * For Cuda and Hip the default resource is *NOT* the CUDA or HIP default stream it is it's 
+            own stream defined in ``camp/include/resource/``. This is in an attempt to break away
+            from some of the issues that arise from the synchronization behaviour of the CUDA/HIP 
+            default stream. It is still possible to use the CUDA/HIP defined default stream as the
+            default resource. This can be enabled by defining ``CAMP_USE_PLATFORM_DEFAULT_STREAM``
+
+------
+Events
+------
+
+Event objects are a feature that allow users to wait or query the status of a resource's action. An event can be returned from a resource with::
+
+    RAJA::resources::Event e = my_res.get_event();
+
+Getting an event like this enqueues an event type object for the given backend. 
+
+You can call a blocking function and wait for that event::
+
+    e.wait();
+
+Preferably, users can enqueue the event to a specific resource, forcing only that resource to wait for the event::
+
+    my_res.wait_for(&e);
+
+The latter is useful as it allows the user to set up dependencies between resource objects and ``RAJA::forall`` calls.
+
+.. note:: *An Event object is only generated if a user specifically returns one from a ``RAJA::forall``::
+           call. This stops unnecessary event objects being created and causing performance hits when not
+           needed. For example::
+    
+               forall<cuda_exec_async<BLOCK_SIZE>>(my_cuda_res, ...
+
+           Will *not* generate a cudaStreamEvent.
+
+                RAJA::resources::Event e = forall<cuda_exec_async<BLOCK_SIZE>>(my_cuda_res, ...
+
+           Will generate a cudaStreamEvent.
+
+-------
+Example
+-------
+
+This example executes three kernels accross two cuda streams on the GPU with a dependence that the first and second kernel finish execution before launching the third. It also demonstrates copying memory from the device to host on a resource:
+    
+First define two concrete CUDA resources and a Host resource::
+
+    RAJA::resources::Cuda dev1;
```

Comment:
This code should be included from the resource example itself (see other examples/tutorial)

Reply:
Yes. @mdavis36 please use the comment syntax in the code example and include the appropriate syntax in the rst file to pull it in. That way if the code changes, the docs will get updated automatically. Here's an example,

https://github.com/LLNL/RAJA/blob/develop/docs/sphinx/user_guide/tutorial/dot_product.rst